### PR TITLE
feat: deploy tenant release v6

### PR DESCRIPTION
**Pull request by benPearce1**
Deploy for specific tenants
```yaml
- task: OctopusDeployReleaseTenanted@6
  name: 'releaseTenants'
  inputs:
    OctoConnectedServiceName: 'octo'
    Space: 'Default'
    Project: 'New Awesome Project'
    ReleaseNumber: '0.0.1'
    Environment: 'test'
    DeployForTenants: |
      tenant 1
      tenant 2
      tenant 4
```

Deploy for tenant tags:
```yaml
- task: OctopusDeployReleaseTenanted@6
  name: 'releaseTenantTags'
  inputs:
    OctoConnectedServiceName: 'octo'
    Space: 'Default'
    Project: 'New Awesome Project'
    ReleaseNumber: '0.0.2'
    Environment: 'test'
    DeployForTenantTags: |
      customer type/early adopter
      maintenance/friday
```

image

### CODE REVIEW

File: source/tasks/DeployTenant/TenantedDeployV6/task.json [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            "helpMarkDown": "Octopus Deploy server connection"
+        },
+        {
+            "name": "Space",
+            "type": "string",
+            "label": "Space",
+            "defaultValue": "",
+            "required": true,
+            "properties": {
+                "EditableOptions": "True"
```
Comment (slewis74) [inline source/tasks/DeployTenant/TenantedDeployV6/task.json:48]: this isn't needed now we aren't using the selectors

File: source/tasks/DeployTenant/TenantedDeployV6/input-parameters.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        variables: variablesMap || undefined,
+        tenants: getLineSeparatedItems(tenantsField || "")?.map((t: string) => t.trim()) || [],
+        tenantTags: tags,
+    };
+
+    const errors: string[] = [];
+    if (parameters.space === "") {
+        errors.push("The Octopus space name is required.");
+    }
+
```
Comment (slewis74) [inline source/tasks/DeployTenant/TenantedDeployV6/input-parameters.ts:76]: we could add some validation here for tenants/tags to help the users. There should be at least 1 tenant or 1 tag.

File: source/tasks/DeployTenant/TenantedDeployV6/task.json [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    "version": {
+        "Major": 6,
+        "Minor": 0,
+        "Patch": 0
+    },
+    "demands": [],
+    "minimumAgentVersion": "2.144.0",
+    "groups": [
+        {
+            "name": "tenant",
```
Comment (slewis74) [inline source/tasks/DeployTenant/TenantedDeployV6/task.json:22]: feels like this probably isn't needed for this step. The groups are good for collapsing optional/advanced things out of the way, but Tenants are required for this step so should always be visible.

**Review by slewis74 — APPROVED** [review]
looks good. A couple of doc/UI thoughts inline for consideration

File: source/tasks/DeployTenant/TenantedDeployV6/task.json (hunk 1 above)
Comment (benPearce1) [reply source/tasks/DeployTenant/TenantedDeployV6/task.json:48]: fixed

File: source/tasks/DeployTenant/TenantedDeployV6/input-parameters.ts (hunk 2 above)
Comment (benPearce1) [reply source/tasks/DeployTenant/TenantedDeployV6/input-parameters.ts:76]: done

File: source/tasks/DeployTenant/TenantedDeployV6/task.json (hunk 3 above)
Comment (benPearce1) [reply source/tasks/DeployTenant/TenantedDeployV6/task.json:22]: group removed, ui screenshot updated

**Pull request merged**